### PR TITLE
dev_guide: improve developing_collections.rst

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
dev_guide: improve developing_collections.rst

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
`docs/docsite/rst/dev_guide/developing_collections.rst`

### CODE REVIEW

**Comment by Andersson007** [comment]
cc @gundalow @felixfontein @acozine @samccann

**Review by gundalow — APPROVED** [review]
From a quick review this looks better, thank you. I think the example of Python imports is much clearer.

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @Andersson007 for the Ansible PR!

**Comment by Andersson007** [comment]
@gundalow @samccann thanks for reviewing and merging!

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 1]
```diff
@@ -714,7 +716,7 @@ If you clone a fork, add the original repository as a remote ``upstream``::
     cd ~/dev/ansible/collections/ansible_collections/community/general
     git remote add upstream git@github.com:ansible-collections/community.general.git
 
-Now you can use this checkout of ``community.general`` in playbooks and roles with whichever version of Ansible you have installed locally, including a local checkout of the ``devel`` branch.
+Now you can use this checkout of ``community.general`` in playbooks and roles with whichever version of Ansible you have installed locally, including a local checkout of the ``main`` branch.
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst:719]: Here `devel` was referring to an Ansible checkout, not to `community.general`. I guess this needs to be formulated in a better way...

Comment (samccann) [reply docs/docsite/rst/dev_guide/developing_collections.rst:719]: gah.. sorry about that. patched it in https://github.com/ansible/ansible/pull/71342

**Pull request merged**